securesystemslib/dsse.py

```diff
@@ -58,21 +58,26 @@ def from_dict(cls, data: dict) -> "Envelope":
         payload = b64dec(data["payload"])
         payload_type = data["payloadType"]
 
-        signatures = [
-            Signature.from_dict(signature) for signature in data["signatures"]
-        ]
+        signatures = []
+        for signature in data["signatures"]:
+            signature["sig"] = b64dec(signature["sig"]).decode("utf-8")
+            signatures.append(Signature.from_dict(signature))
 
         return cls(payload, payload_type, signatures)
 
     def to_dict(self) -> dict:
         """Returns the JSON-serializable dictionary representation of self."""
 
+        signatures = []
+        for signature in self.signatures:
+            sig_dict = signature.to_dict()
+            sig_dict["sig"] = b64enc(sig_dict["sig"].encode("utf-8"))
```

Comment:
I think this is incompatible with other DSSE envelope signing libraries. If I'm reading this right, we're writing the b64 encoding of the hex representation of the signatures. I think we should first decode the hex value before passing the decoded bytes to b64 encoding.

Reply:
Good point, Aditya! DSSE spec doesn't go into details about the format, but I would also expect other implementations to base64-encode the "raw" signature bytes.